### PR TITLE
unittest all and refactored some

### PR DESCRIPTION
**Pull request by stevensnoeijen**
- Replaced the groovy example with example code in the README.md
- Changed the 404 response in the MailChimpClient to throw an error, because getListMember("nonexistingId", "852aaa9532cb36adfb5e9fef7a4206a9") would return null, and this would be confusing.
- Combined most of the MailChimpClient get-calls with query parameters to now be combined in a QueryMap
- added javadoc to all MailChimpClient calls
- Replaced the MailChimpFactory with MailChimpClientBuilder so that mocking Feign would be possible
- Updated to feign 10.1.0 and lombok to 1.18.4
- replaced the old tests with proper UnitTests with mocking

### CODE REVIEW

**Comment by snoein** [comment]
I'm open for suggestions :)

**Comment by snoein** [comment]
bump

File: src/main/java/com/mailchimp/MailChimpClientBuilder.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        this.logger = logger;
+        return this;
+    }
+
+    public MailChimpClientBuilder withBasicAuthentication(String apiKey){
+        this.authRequestInterceptor = new BasicAuthRequestInterceptor("anyString", apiKey);
+        return this;
+    }
+
+    public MailChimpClientBuilder withOAUth2Authentication(String accessToken){
```
Comment (micheal-swiggs) [inline src/main/java/com/mailchimp/MailChimpClientBuilder.java]: Change to `withOAuth2Authentication`.

File: src/main/java/com/mailchimp/domain/List.java [hunk 2]
```diff
@@ -16,13 +18,16 @@
  * @author Micheal Swiggs
  * @author stevensnoeijen
  */
-public class SubscriberList {
+public class List {
```
Comment (micheal-swiggs) [inline src/main/java/com/mailchimp/domain/List.java]: Not to sure about this name. Java already has a `List`. Users will need to start qualifying if using both types in a file.

File: src/main/java/com/mailchimp/domain/Lists.java [hunk 3]
```diff
@@ -1,23 +1,22 @@
 package com.mailchimp.domain;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
-import java.util.List;
 import lombok.Data;
 
 @Data
-public class SubscriberLists {
+public class Lists {
```
Comment (micheal-swiggs) [inline src/main/java/com/mailchimp/domain/Lists.java]: SubscriberLists seems a bit more clearer. The word `Lists` can be both a verb and noun. Also `Lists` by itself gives no indication of what is in the List.

**Comment by micheal-swiggs** [comment]
Nice job with migrating the tests.

File: src/main/java/com/mailchimp/MailChimpClientBuilder.java (hunk 1 above)
Comment (stevensnoeijen) [reply src/main/java/com/mailchimp/MailChimpClientBuilder.java]: good find, fixed it

File: src/main/java/com/mailchimp/domain/List.java (hunk 2 above)
Comment (stevensnoeijen) [reply src/main/java/com/mailchimp/domain/List.java]: I agree and changed it back

**Comment by snoein** [comment]
@micheal-swiggs fixed your requests :)

**Comment by stevensnoeijen** [comment]
bump.
But test (IT) fail because mollie changed their response somewhat and some fail because some id's (like payment) are probably cleaned up. And Travis dont like selenium tests.
What would you suggest?

**Pull request merged**